dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -62,13 +64,63 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 
+    void PushNotificationManager::RegisterUnpackagedApplicationHelper(
+        const winrt::guid& remoteId,
+        _Out_ wil::unique_cotaskmem_string &unpackagedAppUserModelId)
+    {
+        wchar_t processName[1024];
+        THROW_HR_IF(ERROR_FILE_NOT_FOUND, GetModuleFileNameExW(GetCurrentProcess(), NULL, processName, sizeof(processName) / sizeof(processName[0])) == 0);
+
+        THROW_IF_FAILED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+        auto scopeExit = wil::scope_exit(
+            [&]() { CoUninitialize(); });
+
+        wil::com_ptr<INotificationsLongRunningPlatform> notificationPlatform{
+            wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+        winrt::check_hresult(notificationPlatform->RegisterFullTrustApplication(processName, remoteId, &unpackagedAppUserModelId));
+    }
+
+    winrt::hresult PushNotificationManager::CreateChannelWithRemoteIdHelper(const winrt::guid& remoteId, ChannelDetails& channelInfo)
+    {
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+        UINT32 appUserModelIdSize = ARRAYSIZE(appUserModelId);
+
+        THROW_IF_FAILED(GetCurrentApplicationUserModelId(&appUserModelIdSize, appUserModelId));
+
+        THROW_HR_IF(E_INVALIDARG, (appUserModelIdSize > APPLICATION_USER_MODEL_ID_MAX_LENGTH) || (appUserModelIdSize == 0));
+
+        HRESULT operationalCode;
+        ABI::Windows::Foundation::DateTime channelexpirytime{};
+
+        HRESULT hr = PushNotifications_CreateChannelWithRemoteIdentifier(
+            appUserModelId,
+            remoteId,
+            &operationalCode,
+            &channelInfo.channelId,
+            &channelInfo.channelUri,
+            &channelexpirytime);
+
+        /*
+           RemoteId APIs are not applicable for downlevel OS versions.
+           So we get error E_NOTIMPL and we fallback to calling into
+           Public WinRT API CreatePushNotificationChannelForApplicationAsync
+           to request for channels
+        */
+        THROW_HR_IF(hr, (hr != E_NOTIMPL) && (hr != S_OK));
+        THROW_HR_IF(operationalCode, (hr == S_OK) && (operationalCode != S_OK));
+
+        winrt::copy_from_abi(channelInfo.channelExpiryTime, &channelexpirytime);
+        channelInfo.appUserModelId = wil::make_cotaskmem_string(appUserModelId);
+
+        return hr;
```

Comment:
It's weird to see a function both THROW_HR and return hr

Best to pick one error strategy for a function and stick with it